yarn-project/acir-simulator/src/public/index.test.ts

```diff
@@ -81,27 +82,57 @@ describe('ACIR public execution simulator', () => {
         publicContracts.getBytecode.mockResolvedValue(Buffer.from(mintArtifact.bytecode, 'base64'));
 
         // Mock the old value for the recipient balance to be 20
+        const isMinter = new Fr(1n); // 1n means true
         const previousBalance = new Fr(20n);
-        publicState.storageRead.mockResolvedValue(previousBalance);
+        const previousTotalSupply = new Fr(previousBalance.value + 100n);
+        publicState.storageRead
+          .mockResolvedValueOnce(isMinter) // reading whether msg_sender is minter
+          .mockResolvedValueOnce(previousBalance) // reading user's balance
+          .mockResolvedValueOnce(previousTotalSupply); // reading total supply
 
         const execution: PublicExecution = { contractAddress, functionData, args, callContext };
         const result = await executor.simulate(execution, GlobalVariables.empty());
 
-        const expectedBalance = new Fr(160n);
-        expect(result.returnValues[0]).toEqual(expectedBalance);
+        expect(result.returnValues[0]).toEqual(new Fr(1n));
+
+        const recipientBalanceStorageSlot = computeSlotForMapping(new Fr(6n), recipient.toField(), circuitsWasm);
+        const totalSupplyStorageSlot = new Fr(4n);
 
-        const storageSlot = computeSlotForMapping(new Fr(1n), recipient.toField(), circuitsWasm);
+        const expectedBalance = new Fr(previousBalance.value + mintAmount);
+        const expectedTotalSupply = new Fr(previousTotalSupply.value + mintAmount);
+        // There should be 2 storage updates, one for the recipient's balance and one for the total supply
         expect(result.contractStorageUpdateRequests).toEqual([
-          { storageSlot, oldValue: previousBalance, newValue: expectedBalance, sideEffectCounter: 1 }, // 0th is a read
+          {
+            storageSlot: recipientBalanceStorageSlot,
+            oldValue: previousBalance,
+            newValue: expectedBalance,
+            sideEffectCounter: 3,
+          },
+          {
+            storageSlot: totalSupplyStorageSlot,
+            oldValue: previousTotalSupply,
+            newValue: expectedTotalSupply,
+            sideEffectCounter: 4,
```

Comment:
what is `sideEffectCounter`?

Reply:
I am not really sure what it is used for but I think it was introduced when we were dealing with the ordering issue. It counts num reads an writes.

Reply:
Yep, it is used for ordering the side-effects such that a read after a write will ensure you read the updated value etc.